test: add e2e tests for notification items and detail pages

Prithpal-Sooriya opened the pull request:
## **Description**

Adds new e2e test for visiting each notification item we support, and viewing their details pages.

[![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/32799?quickstart=1)

## **Related issues**

Fixes:

## **Manual testing steps**

N/A - run e2e tests.

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**

<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [x] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

Prithpal-Sooriya committed:
test: add e2e tests for notification items and detail pages

Prithpal-Sooriya commented on privacy-snapshot.json:
```diff
@@ -87,5 +87,7 @@
   "user-storage.api.cx.metamask.io",
   "verify.walletconnect.com",
-  "www.4byte.directory"
+  "www.4byte.directory",
+  "i.seadn.io",
+  "images.ctfassets.net"
```
New mocks for images we load during tests.
These are mock notification nft images and announcement assets.

Prithpal-Sooriya commented on test/e2e/mock-e2e-allowlist.js:
```diff
@@ -13,7 +13,6 @@ const ALLOWLISTED_URLS = [
   'https://bafkreifvhjdf6ve4jfv6qytqtux5nd4nwnelioeiqx5x2ez5yrgrzk7ypi.ipfs.dweb.link/',
   'https://bafybeidxfmwycgzcp4v2togflpqh2gnibuexjy4m4qqwxp7nh3jx5zlh4y.ipfs.dweb.link/1.json',
   'https://bridge.api.cx.metamask.io/getTokens?chainId=1',
-  'https://cdn.contentful.com/spaces/jdkgyfmyd9sw/environments/master/entries?content_type=productAnnouncement&order=-sys.createdAt&fields.clients=portfolio',
```
Removing this real-world call, we now utilise mocks.

seaona replied:
great :heart_eyes:  ty for adding mocks for this :pray:

hjetpoluru commented:
@Prithpal-Sooriya, The mock for LIDO_WITHDRAWAL_COMPLETED is not set up properly, which is causing the tests to fail. Could you please take a look?

Prithpal-Sooriya commented:
@hjetpoluru thanks for looking into this!
Yep I will add this to my list on Monday.

hjetpoluru commented on test/e2e/page-objects/pages/notification-details-page.ts:
```diff
@@ -0,0 +1,38 @@
+    try {
+      await this.driver.waitForMultipleSelectors([this.detailsPageBackButton]);
+
+      // Short delay to simulate notification page to load
```
I think may not be needed and used for local testing purpose?

Prithpal-Sooriya replied:
Fixed in https://github.com/MetaMask/metamask-extension/pull/32799/commits/ac3808c27529d352884bd0bec1b8f258f00c9751

Prithpal-Sooriya committed:
refactor: fix eslint issues

Prithpal-Sooriya committed:
Merge branch 'main' of github.com:MetaMask/metamask-extension into MMASSETS-810/add-e2e-tests-for-notification-items

Prithpal-Sooriya committed:
refactor: organise mock notifications

Prithpal-Sooriya committed:
refactor: comment out feature announcement test

Prithpal-Sooriya committed:
refactor: remove unneeded e2e pause

Prithpal-Sooriya committed:
Merge branch 'main' of github.com:MetaMask/metamask-extension into MMASSETS-810/add-e2e-tests-for-notification-items

hjetpoluru committed:
Merge branch 'main' into MMASSETS-810/add-e2e-tests-for-notification-items

Prithpal-Sooriya committed:
Merge branch 'main' of github.com:MetaMask/metamask-extension into MMASSETS-810/add-e2e-tests-for-notification-items

Prithpal-Sooriya committed:
refactor: add back feature announcement test

provide some default fields for controller init, weird how e2e is not providing this

Prithpal-Sooriya commented on app/scripts/controller-init/notifications/notification-services-controller-init.ts:
```diff
@@ -12,8 +12,8 @@ export const NotificationServicesControllerInit: ControllerInitFunction<
         platform: 'extension',
-        spaceId: process.env.CONTENTFUL_ACCESS_SPACE_ID ?? '',
-        accessToken: process.env.CONTENTFUL_ACCESS_TOKEN ?? '',
+        spaceId: process.env.CONTENTFUL_ACCESS_SPACE_ID ?? ':spaceId',
+        accessToken: process.env.CONTENTFUL_ACCESS_TOKEN ?? ':accessToken',
```
Our E2E tests are not populating these environment variables, so we are defaulting to some truthy strings.

Weird how this is not picked up in our GH actions. These are added to CircleCI, so odd how E2E were not picking this up.

hjetpoluru replied:
@Prithpal-Sooriya, I know only for the circleCI environment variables and not the Github actions. I will tag @itsyoboieltr for more details.

Prithpal-Sooriya replied:
Yep lets wait, happy with the current solution 😄 

Thanks both!

hjetpoluru commented on test/e2e/tests/notifications/mocks.ts:
```diff
@@ -1,22 +1,96 @@
+  createMockNotificationLidoStakeCompleted(),
+  createMockNotificationLidoWithdrawalRequested(),
+  createMockNotificationLidoReadyToBeWithdrawn(),
+  createMockNotificationLidoWithdrawalCompleted(),
```
I think this scenario is failing due to the GitHub migration. @Prithpal-Sooriya, should we comment it out for now? Once the migration is complete, we can uncomment it.

Prithpal-Sooriya replied:
Committing suggestion, happy to wait for this change.

Prithpal-Sooriya committed:
Update test/e2e/tests/notifications/mocks.ts

Co-authored-by: Harika <153644847+hjetpoluru@users.noreply.github.com>

Prithpal-Sooriya committed:
refactor: update code suggestion

seaona commented on test/e2e/page-objects/pages/notifications-list-page.ts:
```diff
@@ -11,6 +11,9 @@ class NotificationsListPage {
     '[data-testid="notifications-settings-button"]';
 
+  private readonly notificationListItem = (id: string) =>
+    `[data-testid="notification-list-item-${id}"]`;
```
nice!

seaona commented on test/e2e/tests/notifications/enable-notifications-with-accounts-sync.spec.ts:
```diff
@@ -14,12 +14,76 @@ import {
   this.timeout(120000); // Multiple Syncing features can cause this test to take some time
 
   describe('from inside MetaMask', function () {
+    async function completeOnboardingWithSyncedAccounts(driver: Driver) {
```
just a small nit: it seems like a better place to have this functions might be in a file inside `flows`  folder. This way we'll be able to use them in any test we want. What do you think?
Non-blocking, this could be done in a separate PR if we want to

Prithpal-Sooriya replied:
Wow, I didn't realise this `flows` folder!

If is is okay, since we will add back a commented out test case, I'll do this in a subsequent PR 😄

seaona replied:
thank you @Prithpal-Sooriya ! I've approved the PR then. Could you share the subsequent PR once you have it? :pray:

seaona commented on test/e2e/tests/notifications/enable-notifications.spec.ts:
```diff
@@ -13,10 +12,66 @@ import SettingsPage from '../../page-objects/pages/settings/settings-page';
 
 describe('Enable Notifications - Without Accounts Syncing', function () {
   describe('from inside MetaMask', function () {
+    async function onboardAndAddAccount(driver: Driver) {
```
same goes for these functions here
